### ISSUE TITLE
Index out of Bounds

### ISSUE THREAD

**Issue by stscieisenhamer** [STScI-JWST/jwst#646]
As reported in [Discrepancy Report #003860](https://www.ess.stsci.edu/prsystem/servlet/prbrowse/dr.3860):
```
There was an index out of bounds error for the NIRSpec files listed below.

Files:
jw93045011001_02101_00001_nrs2_rate.fits
jw93045010001_02102_00001_nrs2_rate.fits.
jw93045011001_02103_00001_nrs2_rate.fits
jw84600042001_02101_00001_nrs2_rate.fits
jw84600001001_03101_00001_nrs2_rate.fits
jw93045010001_02103_00001_nrs2_rate.fits
jw93045010001_02101_00001_nrs2_rate.fits
jw93056001001_02101_00001_nrs2_rate.fits


Traceback (most recent call last):
File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/EGG-INFO/scripts/strun", line
14, in step = Step.from_cmdline(sys.argv[1:]) File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/stpipe/step.py", line 128,
in from_cmdline return cmdline.step_from_cmdline(args) File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/stpipe/cmdline.py", line
275, in step_from_cmdline step.run(*positional) File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/stpipe/step.py", line 343,
in run result = self.process(*args) File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py",
 line 166, in process resamp = self.resample_spec(input) File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/stpipe/step.py", line 343,
in run result = self.process(*args) File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/resample/resample_spec_step
.py", line 62, in process good_bits=self.good_bits) File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/resample/resample_spec.py",
 line 92, in __init__ self.build_nirspec_output_wcs() File
"/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb7rc7/lib/python2.7/sit
e-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/resample/resample_spec.py",
 line 210, in build_nirspec_output_wcs dx = xposn[-1] - xposn[0] IndexError:
index -1 is out of bounds for axis 0 with size 0 set savestatus = 1 if ( 1 != 0
) then echo `eval date
'+%Y%j%H%M%S'`-E-ERROR-/ifs/int/jwstd/pipeline/build7/pkgs/miniconda3/envs/jwstb
7rc7/lib/python2.7/site-packages/jwst-0.7.0rc7-py2.7-linux-x86_64.egg/jwst/pipel
ine//calwebb_spec2.cfg failed on jw93045011001_02101_00001_nrs2_rate.fits.
```

**Comment by stscieisenhamer** [comment]
[Discrepancy Report #003861](https://www.ess.stsci.edu/prsystem/servlet/prbrowse/dr.3861) is related but with exposure jw93045010001_02103_00001_nrs2_*

**Comment by stscieisenhamer** [comment]
[Discrepancy Report #003861](https://www.ess.stsci.edu/prsystem/servlet/prbrowse/dr.3861) is related but with exposure jw93045010001_02103_00001_nrs2_*

**Comment by stscieisenhamer** [comment]
@jdavies-st @hbushouse This needs to be resolved, if it hasn't already been through some other issue.

**Comment by jdavies-st** [comment]
These datasets all have the interesting feature of being long slit subarrays.  Currently, they become `ImageModels` after being run through `extract_2d`, which means they don't run through the rest of `Spec2Pipeline` because they don't have a `slits` attribute. `flat_field` and the rest expect `MultiSlitModels` as input.  This is being addressed in #523 .

Of course this issue is about getting them to run through `resample_spec`.  More on that later.

**Comment by hbushouse** [comment]
This has been resolved by various updates in B7.1. Exposures using a single NIRSpec fixed-slit now go through extract_2d and become a MultiSlitModel (containing 1 slit instance), which then goes through resample_spec and extract_1d. Updates to resample_spec have now fixed the error that was happening in B7.0.
